src/Chapter2.hs

```diff
@@ -348,9 +364,8 @@ Implement a function that returns only the first half of a given list.
 >>> firstHalf "bca"
 "b"
 -}
--- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
-
+firstHalf :: [a] -> [a]
+firstHalf l = take ((length l) `div` 2) l
```

Comment:
Yeap!
You can omit spaces before the infixed form function:
```suggestion
firstHalf l = take (length l `div` 2) l
```

Reply:
Thank you 😊